### PR TITLE
Wazo 2393 switchboard moh

### PR DESCRIPTION
**Pull request by pc-m**
Depends-On: https://github.com/wazo-platform/wazo-confd/pull/200

### CODE REVIEW

File: wazo_calld/plugins/switchboards/stasis.py [hunk 1]
```diff
@@ -46,15 +46,20 @@ def notify_all_switchboard_held(self):
...
+                raise Exception('not enough arguments')
+
+            # app_instance = event['args'][0]
+            if event['args'][1] == 'switchboard_queue':
+                self._stasis_start_queue(event_objects, event)
+            elif event['args'][1] == 'switchboard_answer':
+                self._stasis_start_answer(event_objects, event)
+            elif event['args'][1] == 'switchboard_unhold':
+                self._stasis_start_answer_held(event_objects, event)
+        except Exception:
```
Comment (fblackburn1) [inline wazo_calld/plugins/switchboards/stasis.py:60]: Is it possible to use more specific error? RequestException or HTTPError?

File: wazo_calld/plugins/switchboards/confd.py [hunk 2]
```diff
@@ -16,12 +17,24 @@ def __init__(self, tenant_uuid, uuid, confd_client):
...
+        else:
+            return True
+
+    def hold_moh(self):
+        return self._get_queue()['waiting_room_music_on_hold']
+
+    def queue_moh(self):
+        return self._get_queue()['queue_music_on_hold']
+
+    def _get_queue(self):
```
Comment (fblackburn1) [inline wazo_calld/plugins/switchboards/confd.py:32]: I think that you refer to asterisk-queue, but to avoid confusion with wazo-queue
```suggestion
    def _get(self):
```
or
```suggestion
    def _get_switchboard(self):
```

**Review by fblackburn1 — CHANGES_REQUESTED** [review]
I think a small integration tests (without corner case / failing scenario) would be useful and easy (if you can use same logic than test_applications.py)

File: wazo_calld/plugins/switchboards/stasis.py (hunk 1 above)
Comment (pc-m) [reply wazo_calld/plugins/switchboards/stasis.py:60]: I want to catch pretty much everything here.

If an exception is raised during the stasis call handling the call will be stuck in stasis until Asterisk is restarted or the caller gets tired and hangup.

If anything bad happen planned or not the call should be sent back to the dialplan.

File: wazo_calld/plugins/switchboards/confd.py (hunk 2 above)
Comment (pc-m) [reply wazo_calld/plugins/switchboards/confd.py:32]: fixed

**Comment by pc-m** [comment]
> I think a small integration tests (without corner case / failing scenario) would be useful and easy (if you can use same logic than test_applications.py)

I tried and talked about it with @sduthil. The problem is that there is no event when a MOH is started on a bridge, the event is only when the MOH starts on a channel.

To test that the MOH has started would need an asterisk patch or log parsing.

File: wazo_calld/plugins/switchboards/stasis.py (hunk 1 above)
Comment (fblackburn1) [reply wazo_calld/plugins/switchboards/stasis.py:60]: Ok :ok_hand:

**Review by fblackburn1 — DISMISSED** [review]

**Comment by pc-m** [comment]
recheck

**Comment by fblackburn1** [comment]
recheck

**Review by fblackburn1 — DISMISSED** [review]
